### DISCUSSION TITLE
Error might causing duplicate messages

### DISCUSSION THREAD

**Discussion by on-delete** [emqx/emqx#8475] [Q&A] [answered]
This log occured in our EMQX version 4.4.4. We think that it might had something to do with duplicated messages, we received exactly at this time from the EMQX.

`2022-07-13T09:41:57.609586+00:00 [error] client@10.70.1.10:61819 crasher: initial call: emqx_connection:init/4, pid: <0.30109.17>, registered_name: [], error: {{badarity,{#Fun<emqx_shared_sub.1.93993587>,[<46426.3607.0>,no_connection]}},[{emqx_shared_sub,maps_find_by,2,[{file,"emqx_shared_sub.erl"},{line,479}]},{emqx_shared_sub,do_pick,6,[{file,"emqx_shared_sub.erl"},{line,314}]},{emqx_shared_sub,dispatch,5,[{file,"emqx_shared_sub.erl"},{line,140}]},{emqx_broker,do_route,2,[{file,"emqx_broker.erl"},{line,248}]},{emqx_broker,'-route/2-fun-0-',3,[{file,"emqx_broker.erl"},{line,236}]},{lists,foldl,3,[{file,"lists.erl"},{line,1267}]},{emqx_channel,do_publish,3,[{file,"emqx_channel.erl"},{line,608}]},{emqx_connection,with_channel,3,[{file,"emqx_connection.erl"},{line,694}]},{emqx_connection,process_msg,2,[{file,"emqx_connection.erl"},{line,394}]},{emqx_connection,process_msg,2,[{file,"emqx_connection.erl"},{line,400}]},{emqx_connection,handle_recv,3,[{file,"emqx_connection.erl"},{line,358}]},{proc_lib,wake_up,3,[{file,"proc_lib.erl"},{line,236}]}]}, ancestors: [<0.1927.0>,<0.1926.0>,esockd_sup,<0.1584.0>], message_queue_len: 0, messages: [], links: [<0.1927.0>], dictionary: [{{subscribe,<<"topic">>},{allow,1657705284341}},{send_pkt,30290},{incoming_bytes,12532},{{publish,<<"topic">>},{allow,1657705284344}},{guid,{1657705317608707,158806701012381,30289}},{incoming_pubs,84},{'$logger_metadata$',#{clientid => "v4-00000033",peername => "10.70.1.10:61819"}},{recv_msg,30289},{recv_pkt,30291},{'recv_msg.qos1',30289},{outgoing_bytes,181755},{acl_keys_q,{[{publish,<<"topic">>}],[{subscribe,<<"topic">>}]}},{acl_cache_size,2},{rand_seed,{#{bits => 58,jump => #Fun<rand.3.92093067>,next => #Fun<rand.0.92093067>,type => exsss,uniform => #Fun<rand.1.92093067>,uniform_n => #Fun<rand.2.92093067>},[212766467737375251|22254971359885807]}}], trap_exit: false, status: running, heap_size: 4185, stack_size: 28, reductions: 48912652; neighbours:`

Is there something we could take from it, what the cause for the error could be?

**Comment by lafirest** [comment, marked as answer]
Thanks, this is a bug, we will fix it ASAP

**Reply by on-delete** [reply]
Thanks for your quick reply! Then we will wait for the fixed version.

**Reply by on-delete** [reply]
@lafirest Could you give us a link to the issue for this bug, so we could track the status of the fix?
